### PR TITLE
Fixed all warnings and added more pedantic GCC analysis.

### PR DESCRIPTION
**Pull request by szmyd**
Also upgraded to xdi 1.2.1 latest.

### CODE REVIEW

File: src/lib/block/BlockOperations.cpp [hunk 1]
```diff
@@ -646,7 +640,7 @@ void BlockOperations::queuePartialWrite
 void BlockOperations::writeBlobResp
 (
   RequestHandle const&           requestId,
-  WriteBlobResponse const&       resp,
+  WriteBlobResponse const&,
```
Comment (andreasg-fds) [inline src/lib/block/BlockOperations.cpp]: I don't get this change.

Comment (szmyd) [inline src/lib/block/BlockOperations.cpp]: `resp` in the method is unused so generates an unused parameter warning by the compiler. Other than to modify the interface itself not to have this parameter I just removed the name of the variable to indicate to the compiler that I don't intend to use it.

Comment (andreasg-fds) [inline src/lib/block/BlockOperations.cpp]: Cool, that's what I figured once I saw you did it in several places.

**Comment by andreasg-fds** [comment]
:+1:

**Pull request merged**